tools/d2g/d2g.go

```diff
@@ -238,9 +238,11 @@ func artifacts(dwPayload *dockerworker.DockerWorkerPayload) []genericworker.Arti
 	return gwArtifacts
 }
 
-func command(dwPayload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
-	tool := "podman"
-	if dwPayload.Capabilities.Privileged {
+func command(dwPayload *dockerworker.DockerWorkerPayload, dwImage Image, containerEngine string, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
+	tool := containerEngine
+	if payloadDefinedContainerEngine(dwPayload) {
+		tool = dwPayload.Capabilities.ContainerEngine
+	} else if dwPayload.Capabilities.Privileged {
 		tool = "docker"
```

Comment:
I think we should either:
* have a separate worker config for default privileged container engine and default unprivileged container engine to use, or
* we shouldn't switch based on whether the payload is privileged at all (e.g. maybe it is desirable to run `podman --privileged` and using docker always isn't what we want)

For this reason, my instinct says it is better to lose the worker pool setting, keep the task payload setting, and make docker the default, not podman.